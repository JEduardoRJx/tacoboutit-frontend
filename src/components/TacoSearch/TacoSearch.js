import React, { Component } from 'react';
import { Text, View, Dimensions, StyleSheet, Image } from 'react-native';
import { SearchBar } from 'react-native-elements';
import * as Font from 'expo-font';

export default class TacoSearch extends Component {
  state = {
    search: '',
    isLoading: true,
  };
  componentDidMount = async () => {
    await Font.loadAsync({
      'sign-painter': require('../../../assets/fonts/SignPainter-HouseScript.ttf')
    })
    this.setState({ isLoading: false });
  }
  handleChange = (search) => {
    this.setState({ search });
  }
  render() {
    const { search } = this.state;
    const fontFamily = this.state.isLoading ? null : { fontFamily: 'sign-painter' };
    const dynamicInputStyles = [styles.input, fontFamily];
    return (
        <SearchBar
          placeholder="Find Tacos..."
          onChangeText={this.handleChange}
          value={search}
          containerStyle={styles.searchBar}
          inputStyle={dynamicInputStyles}
          placeholderTextColor="black"
          searchIcon={<Image source={require('../../../assets/007-search.png')} style={{width: 45, height: 45}} />}
          inputContainerStyle={styles.inputContainer}
          leftIconContainerStyle={{ marginLeft: 10 }}
        />
    );
  }
}

const styles = StyleSheet.create({
  searchBar: {
    width: Dimensions.get('window').width * .9,
    padding: 0,
    height: 60,
    backgroundColor: 'transparent',
    borderBottomColor: 'transparent',
    borderTopColor: 'transparent'
  },
  input: {
    textAlign: 'center',
    fontSize: 38,
    marginLeft: 0,
    marginTop: 5,
  },
  inputContainer: { 
    backgroundColor: 'white', 
    height: '100%',
    borderRadius: 30,
  },
});

