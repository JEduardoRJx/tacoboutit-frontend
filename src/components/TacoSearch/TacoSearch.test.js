import React from 'react';
import renderer from 'react-test-renderer';

import TacoSearch from '../TacoSearch/TacoSearch';

describe('TacoSearch', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<TacoSearch />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});