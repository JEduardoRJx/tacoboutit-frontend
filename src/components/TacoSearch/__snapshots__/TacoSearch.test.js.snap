// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TacoSearch renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderBottomColor": "transparent",
      "borderBottomWidth": 1,
      "borderTopColor": "transparent",
      "borderTopWidth": 1,
      "height": 60,
      "marginTop": 42,
      "padding": 0,
      "width": 675,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 30,
          "flexDirection": "row",
          "height": "100%",
          "minHeight": 30,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 10,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 45,
              "width": 45,
            }
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Find Tacos..."
        placeholderTextColor="black"
        platform="default"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#86939e",
            "flex": 1,
            "fontSize": 38,
            "marginLeft": 0,
            "marginTop": 5,
            "minHeight": 40,
            "textAlign": "center",
          }
        }
        testID="searchInput"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
            "marginRight": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
